Presentation.Web/app/components/local-config/ui-customization/local-ui-customization.directive.ts

```diff
@@ -0,0 +1,67 @@
+﻿module Kitos.LocalAdmin.Directives {
+    "use strict";
+
+    function setupDirective(): ng.IDirective {
+        return {
+            scope: {
+                customizedModuleId: "@"
+            },
+            controller: LocalUiCustomizationController,
+            controllerAs: "ctrl",
+            templateUrl: `app/components/local-config/ui-customization/local-ui-customization.view.html`
+        };
+    }
+
+    interface IDirectiveScope {
+        /**
+         * The full ui configuration view model
+         */
+        customizationModel: Models.UICustomization.ICustomizedModuleUI | null;
+        /**
+         * Determines the Id of the customization module
+         */
+        customizedModuleId: Models.UICustomization.CustomizableKitosModule;
+        /**
+         * Determines if the editor is open or closed
+         */
+        isOpen: boolean;
+        /**
+         * Update function to toggle a specific setting
+         * @param key
+         */
+        toggleSetting(key : string) : void;
+    }
+
+    class LocalUiCustomizationController implements IDirectiveScope {
+        customizationModel: Models.UICustomization.ICustomizedModuleUI | null = null;
```

Comment:
how does this syntax work?

Reply:
it tells the typescript compiler that this field can either be null or that model. In that way assigning it to anything else will result in and error at compile time